### PR TITLE
RHEL os_version fix

### PR DESCRIPTION
**Pull request by jacknagz**
* There is an issue with the regex for os_version in the `/etc/redhat-release` file, this change properly 
parses version on RHEL6/RHEL7.  
* I also included a new Vagrant box link as chef/centos-6.5 doesn't show up anymore as a valid remote box. 

```
osquery> SELECT * FROM os_version;
+---------------------------------+-------+-------+-------+-------+
| name                            | major | minor | patch | build |
+---------------------------------+-------+-------+-------+-------+
| Red Hat Enterprise Linux Server | 7     | 1     | -1    |       |
+---------------------------------+-------+-------+-------+-------+
```

### CODE REVIEW

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

File: Vagrantfile [hunk 1]
```diff
@@ -7,7 +7,7 @@ targets = {
   },
 
   "centos6.5" => {
-    "box" => "chef/centos-6.5"
+    "box" => "puphpet/centos65-x64"
```
Comment (marpaia) [inline Vagrantfile]: Why the change of the base box?

Comment (jacknagz) [inline Vagrantfile]: It looks like it's non existent upstream anymore:

```
$ vagrant up centos6.5
Bringing machine 'centos6.5' up with 'virtualbox' provider...
==> centos6.5: Box 'chef/centos-6.5' could not be found. Attempting to find and install...
    centos6.5: Box Provider: virtualbox
    centos6.5: Box Version: >= 0
The box 'chef/centos-6.5' could not be found or
could not be accessed in the remote catalog. If this is a private
box on HashiCorp's Atlas, please verify you're logged in via
`vagrant login`. Also, please double-check the name. The expanded
URL and error message are shown below:

URL: ["https://atlas.hashicorp.com/chef/centos-6.5"]
Error: The requested URL returned error: 404 Not Found
```

File: osquery/tables/system/linux/os_version.cpp [hunk 2]
```diff
@@ -22,7 +22,13 @@ namespace xp = boost::xpressive;
...
 namespace tables {
 
-#if defined(REDHAT_BASED)
+#if defined(RHEL)
+const std::string kLinuxOSRelease = "/etc/redhat-release";
+const std::string kLinuxOSRegex =
+    "(?P<name>[\\w+\\s]+) .* "
+    "(?P<major>[0-9]+)\\.(?P<minor>[0-9]+) "
+    "(?P<patch>\\(\\w+\\))";
+#elif defined(CENTOS)
```
Comment (theopolis) [inline osquery/tables/system/linux/os_version.cpp]: change this to `#elif defined(REDHAT_BASED)` to also include the Oracles, Amazon Linuxi. Are you positive this patch only applies to RHEL and is not needed on CentOS?

Comment (jacknagz) [inline osquery/tables/system/linux/os_version.cpp]: Actually, it looks like CentOS doesn't work either.  Can we get rid of the if statement all togehter? 

```
$ cat /etc/redhat-release
CentOS release 6.7 (Final
```

```
osquery> select * from os_version;
+------+-------+-------+-------+-------+
| name | major | minor | patch | build |
+------+-------+-------+-------+-------+
|      | -1    | -1    | -1    |       |
+------+-------+-------+-------+-------+
```

```
[vagrant@localhost ~]$ rpm -qi osquery
Name        : osquery                      Relocations: /
Version     : 1.5.2_57_g65162e7                 Vendor: Facebook
Release     : 1.el6                         Build Date: Tue 15 Sep 2015 01:47:48 AM UTC
...
```

File: Vagrantfile (hunk 1 above)
Comment (sharvilshah) [inline Vagrantfile]: Looks like, [chef moved to centos-6.7](https://vagrantcloud.com/bento/boxes/centos-6.7).
They have old boxes available [here](https://github.com/chef/bento/blob/master/OLD-BOXES.md)

While we are at it, we should also update to their new namespace: `chef` -> `bento`

File: osquery/tables/system/linux/os_version.cpp (hunk 2 above)
Comment (theopolis) [inline osquery/tables/system/linux/os_version.cpp]: Ok, so it works on CentOS 7
```
[reed@localhost] ~/git/github/osquery(branch:master) » ./build/centos7/osquery/osqueryi
E0915 17:11:27.804023  9634 init.cpp:434] Error reading config: config file does not exist
osquery - being built, with love, at Facebook
Using a virtual database. Need help, type '.help'
osquery> select * from os_version;
+--------+-------+-------+-------+-------+
| name   | major | minor | patch | build |
+--------+-------+-------+-------+-------+
| CentOS | 7     | 1     | 1503  |       |
+--------+-------+-------+-------+-------+
osquery> 
[reed@localhost] ~/git/github/osquery(branch:master) » cat /etc/redhat-release
CentOS Linux release 7.1.1503 (Core) 
```

**Comment by theopolis** [comment]
Or change the existing `REDHAT_BASED` regex to:
```cpp
#if defined(REDHAT_BASED)
const std::string kLinuxOSRelease = "/etc/redhat-release";
const std::string kLinuxOSRegex =
    "(?P<name>\\w+) .* "
    "(?P<major>[0-9]+)\\.(?P<minor>[0-9]+)[\\.]{0,1}(?P<patch>[0-9]*).*";
#else
```

**Comment by jacknagz** [comment]
@theopolis That regex doesn't give the full OS name.  Also, I haven't seen usage of the patch in `/etc/redhat-release` for RHEL/CentOS, it's usually `os_name major.minor (release_name)`

**Comment by theopolis** [comment]
Ok, maybe just nix the patch from the `REDHAT_BASED` system's regex.

**Comment by theopolis** [comment]
Hey @jacknagz, can you squash these commits into 1?

**Comment by jacknagz** [comment]
I left in patch, since I have seen some versions of RHEL floating around that are using it.  At most, it will return `-1` if it doesn't exist.

File: osquery/tables/system/linux/os_version.cpp [hunk 3]
```diff
@@ -25,8 +25,9 @@ namespace tables {
 #if defined(REDHAT_BASED)
 const std::string kLinuxOSRelease = "/etc/redhat-release";
 const std::string kLinuxOSRegex =
-    "(?P<name>\\w+) .* "
-    "(?P<major>[0-9]+)\\.(?P<minor>[0-9]+)[\\.]{0,1}(?P<patch>[0-9]+).*";
+    "(?P<name>[\\w+\\s]+) .* "
+    "(?P<major>[0-9]+)\\.(?P<minor>[0-9]+) "
```
Comment (theopolis) [inline osquery/tables/system/linux/os_version.cpp]: Is the " " at the end this ok?

**Comment by marpaia** [comment]
ok to test
